admin_site_search/views.py

```diff
@@ -135,31 +149,60 @@ def match_model(
         return False
 
     def match_objects(
-        self, query: str, model_class: Model, model_fields: List[Field]
+        self,
+        request: HttpRequest,
+        query: str,
+        model_class: Model,
+        model_fields: List[Field],
     ) -> QuerySet:
-        """Returns the QuerySet[:5] after performing an OR filter across all Char fields in the model."""
-        filters = Q()
-
-        for field in model_fields:
-            filter_ = self.filter_field(query, field)
-            if filter_:
-                filters |= filter_
-
-        if filters:
-            results = model_class.objects.filter(filters)[:5]
-        else:
-            results = model_class.objects.none()
-
-        return results
-
-    def filter_field(self, query: str, field: Field) -> Optional[Q]:
-        """Returns a Q 'icontains' filter for Char fields, otherwise None"""
+        """Returns the QuerySet[:5] after performing a search, as per the site_search_method:
+
+        - model_char_fields: OR filter across all Char fields in the model.
+        - admin_search_fields: delegates search to the model's corresponding admin search_fields.
+        """
+        results = model_class.objects.none()
+
+        if self.site_search_method == "model_char_fields":
+            filters = Q()
+
+            for field in model_fields:
+                filter_ = self.filter_field(request, query, field)
+                if filter_:
+                    filters |= filter_
+
+            if filters:
+                results = model_class.objects.filter(filters)
+        elif self.site_search_method == "admin_search_fields":
+            model_admin = self._registry.get(model_class)
+            if model_admin and model_admin.search_fields:
+                results, may_have_duplicates = model_admin.get_search_results(
+                    request=request,
```

Comment:
⚠️ To enable this, f9611145aa44a3227c6be31d0ed56d4589dada69 is a breaking change.

Current thinking: it's useful to have `request` in these methods.

Alternative (non-breaking) option: move this logic inside `def search(self, request)`